modules/c++/nitf/include/nitf/CppField.hpp

```diff
@@ -0,0 +1,362 @@
+/* =========================================================================
+ * This file is part of NITRO
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2017, MDA Information Systems LLC
+ *
+ * NITRO is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef __NITF_CPPFIELD_HPP__
+#define __NITF_CPPFIELD_HPP__
+
+#include <cstdio>
+#include <string>
+#include <str/Convert.h>
+#include <sys/Conf.h>
+#include <nitf/DateTime.hpp>
+#include <nitf/System.hpp>
+
+/*!
+ * File CppField.hpp
+ * Contains pure C++ implemntation for Field
+ */
+namespace nitf
+{
+
+/*!
+ * \class StringConverter
+ * \brief Contains methods to convert arguments to Strings formatted for
+ * NITF Fields
+ */
+class StringConverter
+{
+public:
+    /*!
+     * Construct a StringConverter
+     * \param length How many character should the string be?
+     * \param type NITF_BCS_A or NITF_BCS_N
+     */
+    StringConverter(size_t length, FieldType type) :
+        mLength(length),
+        mFieldType(type)
+    {
+    }
+
+    //! Convert the value to a string of proper length
+    std::string realToString(double value) const;
+    std::string toNitfString(double value) const;
+    std::string toNitfString(float value) const;
+    std::string toNitfString(const DateTime& value) const;
+
+
+    /*!
+     * Convert the value to a string of proper length
+     * \param value The value to convert
+     * \param length How long should the string be? Defaults to mLength.
+     * Intended for internal use.
+     *  \return Formatted string
+     */
+    std::string toNitfString(const std::string& value, size_t length = 0) const;
+
+    template<typename T>
+    std::string toNitfString(const T& value, size_t length = 0) const
+    {
+        if (!std::numeric_limits<T>::is_integer)
```

Comment:
There are some type traits games you could play so that the caller couldn't even call it with a non-integer value (like ConvexHull does it)